src/winc1500/src/winc.c

```diff
@@ -856,9 +868,8 @@ int winc_socket_recv(int fd, uint8_t *buf, uint32_t len, winc_socket_buf_t *sock
             ret = winc_async_request(SOCKET_MSG_RECV, &recv_bytes, timeout);
         }
 
```

Comment:
This is exactly the same as the removed code, so why is this change needed ? Unless you want to check if winc_recv failed first (it's not likely to happen), but you could just check ret first:
```
if (ret != SOCK_ERR_NO_ERROR || recv_bytes <= 0) {
     return (ret != SOCK_ERR_NO_ERROR)? ret: recv_bytes;
}
```